src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -246,16 +316,47 @@ object ApiOperations {
   def fetchAccounts(filter: Filter): Try[Seq[AccountsRow]] =
     fetchLatestBlock().flatMap { latestBlock =>
       Try {
-        val action = for {
-          a: Tables.Accounts <- Tables.Accounts
-          og <- Tables.OperationGroups
-          if a.blockId === latestBlock.hash &&
-          og.source === a.accountId &&
-          filterAccountIDs(filter, a) &&
-          filterAccountDelegates(filter, a) &&
-          filterAccountManagers(filter, a) &&
-          filterOperationIDs(filter, og)
-        } yield (a.accountId, a.blockId, a.manager, a.spendable, a.delegateSetable, a.delegateValue, a.counter, a.script, a.balance)
+        val filteredAccounts = Tables.Accounts.filter({account =>
+            filterAccountIDs(filter, account) &&
+            filterAccountDelegates(filter, account) &&
+            filterAccountManagers(filter, account) &&
+            account.blockId === latestBlock.hash})
+        val filteredOpGroups = Tables.OperationGroups.filter({opGroup =>
+            filterOperationIDs(filter, opGroup) &&
+            filterOperationSources(filter, opGroup) &&
+            filterOperationGroupKinds(filter, opGroup)})
+        val filteredOps = Tables.Operations.filter({op => filterOperationKinds(filter, op)})
+
+        val isOGFilter = isOperationGroupFilter(filter)
+        val isOFilter = isOperationFilter(filter)
+
+        // Join tables in appropriate ways dependents on filters chosen by user
+        val action = (isOGFilter, isOFilter) match {
```

Comment:
It seems this block of code redundantly appears in three functions. Can we just have a single function which takes in the Filter object and returns a tuple with three flags..and then pattern match on that tuple in each of the three calling functions?